PolyFill.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace GrayBMP {
+   class PolyFill {
+      public void AddLine (int x0, int y0, int x1, int y1) => mLines.Add (new (x0, y0, x1, y1));
+
+      public void Fill (GrayBMP bmp) {
+         var width = bmp.Width - 1;
+         for (int i = 0; i < width; i++) {
+            var pts = new List<(int x, int y)> ();
+            foreach (var (x0, y0, x1, y1) in mLines) {
+               if (IsIntersect (x0, y0, x1, y1, 1, i + 0.5, width, out double x, out double y))
+                  pts.Add (new ((int)x, (int)y));
+            }
+            pts = pts.OrderBy (x => x.x).ToList ();
+            for (int j = 0; j < pts.Count; j += 2)
+               bmp.DrawHorizontalLine (pts[j].x, pts[j + 1].x, pts[j].y);
+         }
+      }
+
+      bool IsIntersect (int x0, int y0, int x1, int y1, int x2, double y2, int x3, out double x4, out double y4) {

```

Comment:
Simplify:
- Assume the points are within 0..width, so you don't need to even pass x2 and x3 to this routine (all intersections are valid). 
- We are doing intersections only with horizontal lines. Taking that special consideration, this could be written more simply
- No need to return a y4 value - when we are checking with horizontal lines, y4 will always be equal to y2, right?